### PR TITLE
Added facebook login support

### PR DESCRIPTION
**Pull request by rapee**
### Done
- Added facebook login support
- Added token for API

Support facebook login. For more information, see [API for Facebook Login](https://github.com/youpin-city/youpin-api/pull/154).

On server, authenticated user will be exposed at `req.user` and `res.locals.user`.
On browser, authenticated user will be exposed at `user`.
JWT token, if needed, can be found at `user.token`.
API call on both server and browser are set to automatically use token.

### CODE REVIEW

File: src/assets/views/login.pug [hunk 1]
```diff
@@ -8,32 +8,40 @@ append head
...
       .row
         .col.s12.m8.offset-m2
-
           #login-box.content-padding.opaque-bg
+            .row
+              .col.s12.m8.offset-m2
+                .site-logo.bg
 
-            .site-logo.bg
+                //- h1.page-title= _.startCase(__('action.login'))
```
Comment (supasate) [inline src/assets/views/login.pug]: remove comment

File: src/assets/views/login.pug [hunk 2]
```diff
@@ -8,32 +8,40 @@ append head
...
+                //- form(action='/?login=1', method='GET')
+                //-   .input-field
+                //-     input#login-email(type='email', name='email')
+                //-     label(for='login-email') Email
+                //-   .input-field
+                //-     input#login-password(type='password', name='password')
+                //-     label(for='login-password') Password
+                //-   .input-field
+                //-     input(type='hidden', name='login', value='1')
+                //-     button.btn(type='submit')= __('action.login')
```
Comment (supasate) [inline src/assets/views/login.pug]: remove all comments

File: src/lib/api.js [hunk 3]
```diff
@@ -44,7 +49,14 @@ function APIParser(data) {
   return data;
 }
 
-export default {
-  server: APIFetch(conf.get('site.host'), _.identity, 'text'),
-  api: APIFetch(conf.get('service.api.url'), APIParser)
+// export default {
+//   server: APIFetch(conf.get('site.host'), _.identity, 'text'),
+//   api: APIFetch(conf.get('service.api.url'), APIParser)
+// };
```
Comment (supasate) [inline src/lib/api.js]: remove all comments

File: src/lib/middleware/auth.js [hunk 4]
```diff
@@ -1,43 +1,74 @@
...
-  if (!req.user) {
-    res.redirect('/login');
-    return;
-  }
+  return get_user
+  .then(data => {
+    if (data) {
+      const user = data;
+      user.token = jwt;
+      user.department = data.departments[0];
```
Comment (p16i) [inline src/lib/middleware/auth.js:51]: I think `data.departments[0]` should be `data.department`.

REF : https://github.com/youpin-city/youpin-api/commit/e66db4503dcaf2618b70cd551c1976e9607747a5

Comment (rapee) [reply src/lib/middleware/auth.js:51]: @heytitle You're right

**Comment by rapee** [comment]
If you run `youpin-api` locally, you need to change its config for these fields. I use [MafuENG - Local Web Admin](https://developers.facebook.com/apps/347883645589710/) for testing and already invited you all as admin.

- auth.frontendSuccessRedirect: http://localhost:8080/auth/success (`youpin-web-admin` domain)
- auth.frontendFailureRedirect: http://localhost:8080/auth/fail (`youpin-web-admin` domain)
- auth.facebook.clientID: 347883645589710
- auth.facebook.clientSecret: 88aff10c4bedd5b661421044c6619209

**Comment by supasate** [comment]
`frontendSuccessRedirect` and `frontendFailureRedirect` are deprecated now. User the following keys instead:
- auth.userSuccessRedirect
- auth.userFailureRedirect
- auth.staffSuccessRedirect
- auth.staffFailureRedirect

**Pull request merged**